iml-agent/src/daemon_plugins/ntp.rs

```diff
@@ -0,0 +1,536 @@
+// Copyright (c) 2019 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::{
+    agent_error::ImlAgentError,
+    cmd::cmd_output,
+    daemon_plugins::{DaemonPlugin, Output},
+    systemd,
+};
+use futures::{future, future::IntoFuture, Future, Stream};
+use iml_fs::stream_file_lines;
+use iml_util::Flatten;
+use iml_wire_types::AgentResult;
+use lazy_static::lazy_static;
+use parking_lot::RwLock;
+use regex::Regex;
+use serde_json::json;
+use std::time::{SystemTime, UNIX_EPOCH};
+use std::{path::Path, process::Command};
+use std::{process, sync::Arc};
+use tokio_process::CommandExt;
+
+static NTP_CONFIG_FILE: &'static str = "/etc/ntp.conf";
+
+type BoxedFutureString = Box<dyn Future<Item = String, Error = ImlAgentError> + Send>;
+
+#[derive(Debug)]
+pub struct Ntp;
+
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+pub struct SystemdUnit {
+    enabled: bool,
+    active: bool,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct TimeSyncServices {
+    ntpd: SystemdUnit,
+    chronyd: SystemdUnit,
+}
+
+impl TimeSyncServices {
+    fn is_ntp_only(&self) -> bool {
+        self.ntpd.enabled && self.ntpd.active && !self.chronyd.enabled && !self.chronyd.active
+    }
+}
+
+#[derive(Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+pub enum TimeSync {
+    Synced,
+    Unsynced,
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+pub struct TimeStatus {
+    synced: TimeSync,
+    offset: Option<String>,
+}
+
+fn is_ntp_synced(output: String) -> TimeSync {
+    match output.as_ref() {
+        "b true" => TimeSync::Synced,
+        "b false" => TimeSync::Unsynced,
+        _ => TimeSync::Unsynced,
```

Comment:
Yes of course. The manager plugin will be monitoring the results and if it gets Unsynced it will generate a warning on the manager side. I'm working on piece now.

Reply:
I meant like a trace::warning() to log an unexpected value,

Reply:
Added a warning.